### PR TITLE
Fix: implement clarity word pattern

### LINKED ISSUE #599: autocomplete takes only the word after the last '-'
for reference, the flow when using the autocomplete:
1. write part of first word and autocomplete everything with enter
2. directly overwrite the second word
3. press tab and overwrite the third word

problem case: 
I wrote `define-private`, wanted to change to `define-public` cause it was miss picked when fast tabbing 
press ctrl-z to undo and looks like in the 2nd line, then added letter `u` and tabbed and got this 

```
(define-private func-name body)
define-p
define-(define-public func-name body)
```

p.s. it is because of it picking `define-public` just from the word `pu` not `define-pu` which is counterintuitive

### PR DESCRIPTION
**Pull request by hugocaillard**
Fixes #599 

On hover, VSCode better understand keywords as well.

Before:
Screenshot 2022-10-19 at 16 21 55
After:
Screenshot 2022-10-19 at 16 22 21

@obycode you're the expert, I've seen some language doing way more complex regexps but this PR seems to do the trick

### CODE REVIEW

File: components/clarity-vscode/language-configuration.json [hunk 1]
```diff
@@ -24,5 +24,6 @@
     ["(", ")"],
     ["\"", "\""],
     ["'", "'"]
-  ]
+  ],
+  "wordPattern": "[\\w\\d-_!?]+"
```
[the remark this answers is not in the document]
Comment (hugocaillard) [reply components/clarity-vscode/language-configuration.json]: Right, done ✅ 
Can you think of any missing characters?

[the remark this answers is not in the document]
Comment (obycode) [reply components/clarity-vscode/language-configuration.json]: Oh yeah, it should also include `+<>=/*`.

[the remark this answers is not in the document]
Comment (obycode) [reply components/clarity-vscode/language-configuration.json]: Those are, unfortunately, allowed.

[the remark this answers is not in the document]
Comment (hugocaillard) [reply components/clarity-vscode/language-configuration.json]: Done @obycode

**Review by obycode — APPROVED** [review]
LGTM!

**Pull request merged**